src/NumSharp.Core/Logic/np.logical_and.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using NumSharp.Backends;
+using NumSharp.Generic;
+using NumSharp.Utilities;
+
+namespace NumSharp
+{
+    public static partial class np
+    {
+        /// <summary>
+        ///     Compute the truth value of x1 AND x2 element-wise.
+        /// </summary>
+        /// <param name="lhs">Input boolean array.</param>
+        /// <param name="rhs">Input boolean array.</param>
+        /// <returns>Returns True if the arrays are equal.</returns>
+        /// <remarks>https://docs.scipy.org/doc/numpy/reference/generated/numpy.logical_and.html</remarks>
+        public static unsafe NDArray<bool> logical_and(NDArray lhs, NDArray rhs)
+        {
+            if (lhs.typecode == NPTypeCode.Boolean && rhs.typecode == NPTypeCode.Boolean)
```

Comment:
Logical And can happen with any types. not just boolean vs boolean. please handle the rest of the cases. I would suggest using Regen.

Reply:
I'd say we only handl bool is good.
https://docs.scipy.org/doc/numpy/reference/generated/numpy.logical_or.html

![image](https://user-images.githubusercontent.com/1705364/72763199-35b7bd80-3ba8-11ea-9379-90fe9ba73df8.png)

Reply:
```python
import numpy as np
>>> np.logical_and(np.arange(10), np.arange(10))
[False  True  True  True  True  True  True  True  True  True]
```
Behind the scenes the np.arange(10) is casted to ndarray of bool by `x != 0`.
This should be done to all dtypes.